code/frameworks/svelte-vite/src/plugins/generateDocgen.ts

```diff
@@ -0,0 +1,437 @@
+  switch (expr.kind) {
+    case ts.SyntaxKind.TrueKeyword:
+      return { text: 'true' };
+    case ts.SyntaxKind.FalseKeyword:
+      return { text: 'false' };
+    case ts.SyntaxKind.NullKeyword:
+      return { text: 'null' };
+  }
+  return undefined;
+}
+
+function loadCompilerOptions(basepath: string): ts.CompilerOptions {
+  const jsConfigPath = ts.findConfigFile(basepath, ts.sys.fileExists, 'jsconfig.json');
+  const tsConfigPath = ts.findConfigFile(basepath, ts.sys.fileExists);
+  const configPath = jsConfigPath || tsConfigPath;
+
+  const forcedOptions = {
+    sourceMap: false,
+    noEmit: true,
+    allowJs: true,
+    checkJs: true,
+    skipLibCheck: true,
+    skipDefaultLibCheck: true,
+  };
+  if (!configPath) {
+    return forcedOptions;
+  }
+
+  const configFile = ts.readConfigFile(configPath, ts.sys.readFile);
+  const config = ts.parseJsonConfigFileContent(
+    configFile.config,
+    ts.sys,
+    path.dirname(configPath),
+    undefined,
+    configPath,
+    undefined
+  );
+
+  return {
+    ...config.options,
+    ...forcedOptions,
+  };
+}
+
+export interface SourceFileCache {
+  filenameToSourceFiles: Record<string, ts.SourceFile>;
+  hashToSourceFiles: Record<string, ts.SourceFile>;
+}
+
+export function createSourceFileCache(): SourceFileCache {
+  return {
+    filenameToSourceFiles: {},
+    hashToSourceFiles: {},
+  };
+}
+
+export function generateDocgen(targetFileName: string, sourceFileCache: SourceFileCache): Docgen {
+  if (targetFileName.endsWith('.svelte')) {
+    targetFileName = targetFileName + '.tsx';
+  }
+  let propsRuneUsed = false;
+
+  const options = loadCompilerOptions(targetFileName);
+
+  // Create a custom host
+  const originalHost = ts.createCompilerHost(options);
+  const host: ts.CompilerHost = {
+    ...originalHost,
+    fileExists(fileName) {
+      let exists = originalHost.fileExists(fileName);
+      if (exists) {
+        return exists;
+      }
+
+      if (fileName.endsWith('.svelte.tsx') || fileName.endsWith('.svelte.jsx')) {
+        fileName = fileName.slice(0, -4); // remove .tsx or .jsx
+        exists = originalHost.fileExists(fileName);
+        return exists;
+      }
+
+      return false;
+    },
+    getSourceFile(fileName, languageVersion, onError) {
+      if (fileName.endsWith('.svelte.tsx') || fileName.endsWith('.svelte.jsx')) {
+        // .svelte file (`import ... from './path/to/file.svelte'`)
+
+        const realFileName = fileName.slice(0, -4); // remove .tsx or .jsx
+        const content = originalHost.readFile(realFileName);
+        if (content === undefined) {
+          return;
+        }
+
+        const digest = crypto.createHash('sha256').update(content).digest('hex');
+
+        if (sourceFileCache.hashToSourceFiles[digest]) {
+          return sourceFileCache.hashToSourceFiles[digest];
+        }
+
+        const tsx = svelte2tsx.svelte2tsx(content, {
+          version: VERSION,
+          isTsFile: true,
+          emitOnTemplateError: true,
```

Comment:
This option currently is to try to preprocess the code so the svelte compiler won't complain that much. But it's mostly only used in the editor to provide code completion. It's only a string check without the reliance on a parser so it only prevents simple cases.  Not sure if this is needed since the build tool might complain anyway.

Reply:
Thanks. I just removed this line!